manifest.go

```diff
@@ -1480,6 +1485,33 @@ func avroPartitionData(input map[string]any, nameToID map[string]int, logicalTyp
 	return out
 }
 
+func avroFieldPartitionData(input map[int]any, idToName map[int]string, logicalTypes map[int]avro.LogicalType) map[int]any {
```

Comment:
the `idToName` map isn't being used at all, we should probably remove that as an argument